nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/encrypt/CipherPropertyEncryptor.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.encrypt;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.nifi.security.kms.CryptoUtils;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.Security;
+
+/**
+ * Cipher Property Encryptor provides hexadecimal encoding and decoding around cipher operations
+ */
+abstract class CipherPropertyEncryptor implements PropertyEncryptor {
+    private static final Charset PROPERTY_CHARSET = StandardCharsets.UTF_8;
+
+    static {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    /**
+     * Encrypt property and encode as a hexadecimal string
+     *
+     * @param property Property value to be encrypted
+     * @return Encrypted and hexadecimal string
+     */
+    @Override
+    public String encrypt(final String property) {
+        final byte[] binary = property.getBytes(PROPERTY_CHARSET);
```

Comment:
Are NiFi properties always UTF_8 encoded regardless of language?

Reply:
That's a good question.  This approach preserves the current behavior and makes it more explicit.  `StringEncryptor` previously buried this fact in the `encryptKeyed` and `encryptPBE` methods when calling `Cipher.doFinal()`.